new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -324,7 +342,19 @@ public void testCreateMiiDomain() {
           managedServerPrefix + i, domainNamespace);
       checkServiceCreated(managedServerPrefix + i, domainNamespace);
     }
-
+    
+    // check and wait for the app to be accessible in all server pods
+    for (int i = 1; i <= replicaCount; i++) {
+      checkAppRunning(
```

Comment:
We don't want to rely on a load balancer here, and the managed servers don't have a NodePort.

Reply:
I'm confused now too.  We discussed this in the test meeting today.  I thought I heard something different.  That is, the use case is not a redeploy, but instead just a cluster rolling.  Since the application is available on all cluster members, then the application would be continually available through a load balancer.  This isn't the same thing as the "continuous availability" option of course, but the application should be available through the load balancer all of the time because there will always be at least one cluster member ready.  This check looks like something different.

Reply:
This test class contains 5 test cases in this PR. We are at the end of the first end-to-end test case testCreateMiiDomain, which does not do any patching/rolling-restart. The checkAppRunning method here just checks and waits util the application is accessible on all server pods as a sanity check of a stable domain/cluster. It fails if the app is not available on all servers within the given period of time. The check is performed by running curl command inside each pod.

Reply:
The #4 and #5 test cases added in this PR, namely testPatchAppV2 and testAddSecondApp, are the use cases where we patch the domain resource with a new image that contains a newer version of an existing app, and adds a new app,  respectively. Only in testPatchAppV2, we check the application "continuous" availability. We achieve this by starting a thread to collect the availability data of the app (V1 or V2)while the main thread does the patching operation and regular verification. At the end when the V2 app becomes available on all servers, we use the collected data to determine if there are app outage during the period of time when the patching and rolling-restart are in progress.